source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -84,91 +80,106 @@ namespace aspect
+            const auto particle_property_value = particle->get_properties();
+            for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+              if (selected_properties[property_index])
+                r[property_index][positions_index] = particle_property_value[property_index];
 
             const Tensor<1, dim, double> relative_particle_position = (particle->get_location() - approximated_cell_midpoint) / cell_diameter;
-            A(index,0) = 1;
-            A(index, 1) = relative_particle_position[0];
-            A(index, 2) = relative_particle_position[1];
+            A(property_index, 0) = 1;
+            A(property_index, 1) = relative_particle_position[0];
+            A(property_index, 2) = relative_particle_position[1];
             if (dim == 2)
               {
-                A(index, 3) = relative_particle_position[0] * relative_particle_position[1];
+                A(property_index, 3) = relative_particle_position[0] * relative_particle_position[1];
               }
             else
               {
-                A(index, 3) = relative_particle_position[2];
-                A(index, 4) = relative_particle_position[0] * relative_particle_position[1];
-                A(index, 5) = relative_particle_position[0] * relative_particle_position[2];
-                A(index, 6) = relative_particle_position[1] * relative_particle_position[2];
-                A(index, 7) = relative_particle_position[0] * relative_particle_position[1] * relative_particle_position[2];
+                A(positions_index, 3) = relative_particle_position[2];
+                A(positions_index, 4) = relative_particle_position[0] * relative_particle_position[1];
+                A(positions_index, 5) = relative_particle_position[0] * relative_particle_position[2];
+                A(positions_index, 6) = relative_particle_position[1] * relative_particle_position[2];
+                A(positions_index, 7) = relative_particle_position[0] * relative_particle_position[1] * relative_particle_position[2];
               }
           }
 
         dealii::LAPACKFullMatrix<double> B(matrix_dimension, matrix_dimension);
 
-        Vector<double> c_ATr(matrix_dimension);
-        Vector<double> c(matrix_dimension);
+        std::vector<Vector<double>> c_ATr(n_particle_properties, Vector<double>(matrix_dimension));
+        std::vector<Vector<double>> c(n_particle_properties, Vector<double>(matrix_dimension));
 
-        const double threshold = 1e-15;
+        constexpr double threshold = 1e-15;
         unsigned int index_positions = 0;
 
-        // Form the matrix B=A^TA and right hand side A^Tr of the normal equation.
-        A.Tmmult(B, A, false);
-        A.Tvmult(c_ATr,r);
+        // Matrix A can be rank deficient if it does not have full rank, therefore singular.
+        // To circumvent this issue, we solve A^TAx=A^Tr by using singular value
+        // decomposition (SVD).
```

Comment:
This comment was replaced on the master branch in #3394 (probably after you created your branch) by:
`// Form the matrix B=A^TA and right hand side A^Tr of the normal equation.` Please use the new version unless there is value in having the extended description (if so, please reword to form a more concise comment).